plugins/widget/plugin.js

```diff
@@ -3605,8 +3620,13 @@
 				partName;
 
 			for ( partName in widget.parts ) {
+				if ( typeof widget.parts[ partName ] !== "string" ) {
+					parts[ partName ] = widget.parts[ partName ];
+					continue;
+				}
+
 				el = widget.wrapper.findOne( widget.parts[ partName ] );
-				parts[ partName ] = el;
+				parts[ partName ] = el ? el : widget.parts[ partName ];
```

Comment:
The CI is red (surprisingly a lot of tests started to fail) and it seems this line is problematic one. My assumption is that other widget functions and different plugins checks parts name for `null`. If it's not `null` they treat it as an `element` and try to perform typical elements operations on it.

It looks like a breaking change TBH and we can't change it easily. The only reasonable solution I'm able to think of as for now is to keep copy of initial `widget.parts` and use it for refreshing parts which weren't present during initialization :thinking: 

The only issue is that `widget.parts` is a part of a public API (see https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_plugins_widget_definition.html#property-parts) so probably one may try to change it directly via `widgetInstance.parts = ...` but I don't think we will have such case. We should just remember to mention in the `refreshParts` method docs that it works on initial parts and wouldn't cover a case when `parts` were updated manually after widget initialization.

Reply:
As you suggested I created a new object called `partSelectors` which keeps original CSS selectors for parts and is used to reinitialize parts if they are refreshed.
Also we agreed to create a flag for user to decide if already existing parts should be reinitialized, but TBH I don't remember what should be default option 😄 I set it to `false` (so existing parts should be ignored), but I can changed it if we agreed otherwise.

Reply:
I think we agreed that by default it should refresh all parts and the flag will be a safe switch for some specific use cases when one needs to only refresh not yet initialized parts.

Reply:
Thanks, so I'll change it in a moment 👍